### ISSUE TITLE
Trouble connecting to server

### ISSUE THREAD

**Issue by nichoth** [logux/logux#118]
Thanks for your help. My feeling is that this is a very simple issue that I am blind to.

I am starting a new project, creating a client in a browser:
```js
    const client = new CrossTabClient({
        server: 'ws://localhost:8765',
        subprotocol: '1.0.0',
        userId: '123',
        token: 'abc'
    })
```

The server is a local node:
```js
const server = new Server(
    Server.loadOptions(process, {
        subprotocol: '1.0.0',
        fileUrl: import.meta.url,
        supports: '1.x',
        port: 8765
    })
)
```

This issue is frustrating because I am getting no output at all. I see in the browser it says "No Internet connection", and there is nothing logged on the server. So the browser is not seeing the server at all.

This was working just recently in the past, I would see logs on the server like "bad authentication". It stopped working, and I do not know what changed, and cannot see what could be wrong.

I have made this test public, if anyone is interested in seeing more in more detail: https://github.com/nichoth/try-logux/blob/wip/src/state/index.ts#L34

**Comment by ai** [comment]
Very-very strange. Maybe you disable Internet in browser’s DevTools?

**Comment by nichoth** [comment]
Thanks for the response. I thought that too, but no, network is all ok in dev tools.

<img width="697" alt="image" src="https://github.com/logux/logux/assets/5776508/d07456d5-d808-483f-a456-1f8922f21bd4">

That is a screenshot of websocket connections in the `network` tab. We see 1 connection, which I think is for `vite`, nothing for logux.

In terminal, we see that `npm run start:logux` works ok:

<img width="682" alt="image" src="https://github.com/logux/logux/assets/5776508/58e58821-c9e4-419a-b063-caa3b4859737">

But it is not seeing any connection attempts. The URL `http://127.0.0.1:8765/health` is ok too; the server seems to be working.

**Comment by nichoth** [comment]
Yes, the browser side is not connecting. I'm not sure why though. In the browser, we call `client.start()` is there something else I should do?

<img width="571" alt="image" src="https://github.com/logux/logux/assets/5776508/f2c2ff17-62d4-4950-8e33-871f396b9955">

**Comment by ai** [comment]
No, your code looks correct.

Do you have another machine or another browser to try? (Sorry I am busy on another open source project to try it on my machine).

**Comment by nichoth** [comment]
Ok, I think I've got it. Thanks for looking at this with me. The issue seems to be that I had  multiple tabs open to the same `localhost:8888` address.

When the new client tab would open, we see it has a `role` of `follower`, but if I wait for a while, maybe ~1 minute, then I do see some connection attempts logged by the server

<img width="682" alt="image" src="https://github.com/logux/logux/assets/5776508/e3a2b666-047c-4a7c-a2a3-0e8bb9545cbb">

When I see the connection attempts in the server log, we see that the client has changed to `role` = `'leader'`.

<img width="570" alt="image" src="https://github.com/logux/logux/assets/5776508/5c7dda24-0515-4e74-8d7e-87b85b6aaa18">

-------

So I closed all tabs, and now it works as expected — starts with role of 'leader', and we instantly see connection attempts.

<img width="579" alt="image" src="https://github.com/logux/logux/assets/5776508/1c176976-88b5-451a-858f-9626457751e3">

Sorry, my gut feeling was correct here, and this was a very simple but hard to find issue. Thanks for checking with me.
